src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferenceCache.cs

```diff
@@ -57,197 +48,156 @@ public static SymbolInfo GetSymbolInfo(SemanticModel model, SyntaxNode node, Can
             }
 
             if (entry.AliasNameSet.Contains(token.ValueText))
-            {
-                return model.GetAliasInfo(token.Parent, cancellationToken);
-            }
+                return model.GetAliasInfo(token.GetRequiredParent(), cancellationToken);
 
             return null;
         }
 
-        public static ImmutableArray<SyntaxToken> GetIdentifierTokensWithText(
-            ISyntaxFactsService syntaxFacts,
-            SemanticModel model,
-            SyntaxNode root,
-            SourceText sourceText,
-            string text,
+        public static async Task<ImmutableArray<SyntaxToken>> FindMatchingIdentifierTokensAsync(
+            Document document,
+            SemanticModel semanticModel,
+            string identifier,
             CancellationToken cancellationToken)
         {
-            var normalized = syntaxFacts.IsCaseSensitive ? text : text.ToLowerInvariant();
```

Comment:
note: the normalized value isn't necessary.  we store this as the key in a dictionary.  so we instead now use an appropriate EqualityComparer for that dictionary isntead.